src/compat/posix/include/sys/mman.h

```diff
@@ -28,27 +30,20 @@ __BEGIN_DECLS
 
 #define MAP_FAILED (void*)-1
 
-#if 1
-#include <errno.h>
-static inline void  *mmap(void *addr, size_t len, int prot, int flags, int fd, off_t off) {
-	// ToDo: implement for InitFS files
-	(void)addr;
-	(void)len;
-	(void)prot;
-	(void)flags;
-	(void)fd;
-	(void)off;
-	//printf(">>> mmap(%i)\n",fd);
-	errno = EPERM;
-	return NULL;
-}
-
+extern struct page_allocator *__phymem_allocator;
+extern int    mprotect(void *, size_t, int);
 
+// todo: implement for InitFS files
+static inline void  *mmap(void *addr, size_t len, int prot, int flags, int fd, off_t off) {
+	size_t full_pages = len < PAGE_SIZE() ? 1 : len / PAGE_SIZE();
+	size_t pages = len % full_pages ? full_pages + 1 : full_pages;
 
-#endif
+	// todo: check the standard for zero initialization
+	void *ptr = flags & MAP_ANONYMOUS ? page_alloc_zero(__phymem_allocator, pages) : NULL;
```

Comment:
The operator " ? : " can be handy sometimes, but IMHO in this case it makes code less understandable.

You can rewrite it like
`if (flags & MAP_ANONYMOUS) {`
`    ...`
`} else {`
`    SET_ERRNO(ENOTSUP);`
`    return NULL;`
`}`

However, it's not critical.